test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -52,7 +52,7 @@
  * @test
  * @library /test/lib
  * @build jdk.test.lib.RandomFactory
- * @run testng TransferTo
+ * @run testng/othervm TransferTo
```

Comment:
Hello Brian, should this also perhaps set the `-Xmx` value to a higher value or is the current VM defaults for max heap size on all VMs expected to be big enough for this test?

Reply:
> Hello Brian, should this also perhaps set the `-Xmx` value to a higher value or is the current VM defaults for max heap size on all VMs expected to be big enough for this test?

This may be helpful as this test needs to do some big transfer. For the specific OOME sighting then I suspect it may be more to do with how the tests are run. From what I can tell, it's running a very large group of tests and it's likely that the agent VM has been running for a long time. It could be that other tests that ran previously in this VM have left things behind.

Reply:
The version of the test proposed here passes with the command line option `-Xmx512M`, so if the default maximum heap size is really half the physical memory then I don't think it is necessary to set `-Xmx`.

Reply:
on the machine env where the failure has occurred we see that JVM has default heap size at 30GB and the machine has physical memory at 762GB

Reply:
So we would not expect an OOME for the test unless something else were gobbling up the heap memory.

Reply:
I've just checked the jtreg parameters for vm options and the -Xmx512M is set for the test
also sets a I've just checked and the test is not run with default option, but with explicit setting of 512MB
a MAX_RAM_PERCENTAGE option

-vmoption:-XX:MaxRAMPercentage=$MAX_RAM_PERCENTAGE 

MAX_RAM_PERCENTAGE == 25/18 which rounds up to 1